docs: changelog and upgrade (#103)
shopware/SwagMigrationAssistant: CHANGELOG.md

# 16.0.0

> - #11808 - Introduced Error Resolution to manage data inconsistencies and apply migration fixes directly in the administration.

Reviewer: ```suggestion
- #11808 - Introduced Error Resolution to manage data inconsistencies and apply migration fixes directly in the SW6 administration, instead of fixing issues in the source system as it was needed previously.
```

- #11808 - Enriched migration logs with detailed data descriptions to improve error traceability.
- #11808 - Integrated ACL constraints to restrict specific migration actions based on user permissions.
- #11808 - Improved data reliability via stronger validation checks for converted data.

> - #11808 - Applied general UI/UX improvements for an overall visual consistency.

Reviewer: ```suggestion
- #11808 - Applied general UI/UX improvements for visual consistency.
```

> - #11808 - Removed CLI commands `migration:abort`, `migration:get-progress`, and `migration:start` in favor of message queue based execution.

Reviewer: These were already using "message based execution" 😅 
```suggestion
- #11808 - Removed CLI commands `migration:abort`, `migration:get-progress`, and `migration:start`. You have to start the migration via the Administration UI, it still runs in the background message queue
```

- MIG-1046 - Fixed an issue related to the validation of json fields for a migration run, which could prevent starting a migration in the latest shopware version

# 13.1.0

- MIG-981 - Media migration now uses the system's temporary directory to download files
- MIG-1016 - Improves the warnings for different default currency and default language in the data selection
- MIG-1016 - Added new block `{% block swag_migration_confirm_warning_alert %}` in `swag-migration/component/card/swag-migration-confirm-warning/swag-migration-confirm-warning.html.twig`
- MIG-1037 - Fixes a rare issue that in certain situations not all entities are migrated (some were skipped). Was detected during translations of SW5

# 13.0.0

- MIG-945 - [BREAKING] Changed method name `getMedia` to `setMedia` in `SwagMigrationAssistant\Profile\Shopware\Converter\PropertyGroupOptionConverter`
- MIG-945 - [BREAKING] Removed cli command `migration:migrate` use `migration:start` instead
- MIG-945 - [BREAKING] Changed method `writePremapping` of `SwagMigrationAssistant\Controller\PremappingController`
  - Changed return type from `JsonResponse` to `Response`

>   - Removed parameter `runUuid`

Reviewer: Markdown is weird, because there are many flavors of it. What's your reasoning for changing this from four spaces to two?

Normal markdown suggests four spaces:
https://www.markdownguide.org/basic-syntax/#adding-elements-in-lists

GH markdown needs it visually aligned with first character of parent (which would be two in this case):
https://docs.github.com/en/get-started/writing-on-github/getting-started-with-writing-and-formatting-on-github/basic-writing-and-formatting-syntax#nested-lists

but in practice it doesn't seem to matter for GH and it also accepts four spaces and renders it the same:

```md
- a
    - b
- a
  - b
```

- a
    - b
- a
  - b

So I would suggest sticking two four spaces, as that was working fine in our extension store changelog: https://store.shopware.com/en/swag257162657297f/migration-assistant.html

I'm also fine with two spaces, which works fine for GH and in PHPStorm, but don't know the markdown flavor / lib used in our extension store and if it might run into issues with that 🤷‍♂️ . I mean we could try 😅

Author: will change it back to 4

- MIG-945 - [BREAKING] Removed property `fileChunkByteSize` of `SwagMigrationAssistant\Migration\MessageQueue\MessageProcessMediaMessage`
- MIG-945 - [BREAKING] Removed property `runRepo` of `SwagMigrationAssistant\Migration\Service\PremappingService`
- MIG-945 - [BREAKING] Changed `SwagMigrationAssistant\Migration\Service\PremappingServiceInterface` / `SwagMigrationAssistant\Migration\Service\PremappingService`
  - Removed parameter `run` of method `generatePremapping`

>   - Added parameter `dataSelectionIds` to method `generatePremapping`

Reviewer: see above and all the other places where this indentation was changed
